Next Release

mohammadranjbarz opened the pull request.

Meriem-B committed:
fix: add pagination for qf round listed projects

Meriem-B committed:
fix: conflict

Meriem-B committed:
fix: Refactor conditions (use optinal chaining)

Meriem-B committed:
fix: add alternative text for accessibility in SVG elements.

Meriem-B committed:
fix: add detailed func attribute types

Meriem-B committed:
fix: add attributes types to handleBannerBgImage func

Meriem-BM committed:
Merge pull request #1597 from Giveth/fix/unlink_project_with_qf_round

fix: 1125 issue with unlinking project from qf round

Meriem-BM committed:
Merge branch 'staging' into fix/add_qf_round_pagination

Meriem-BM committed:
Merge pull request #1601 from Giveth/fix/add_qf_round_pagination

fix: add pagination for QF Round listed projects

RamRamez committed:
remove refreshProjectDonationSummaryView from migrations

RamRamez committed:
DropDonationSummaryView migration

RamRamez committed:
remove ProjectDonationSummaryView

RamRamez committed:
remove refreshProjectDonationSummaryView

RamRamez committed:
replace countUniqueDonors and sumDonationValueUsd with countUniqueDonorsAndSumDonationValueUsd

RamRamez committed:
remove ProjectDonationSummaryView

RamRamez committed:
Merge pull request #1603 from Giveth/drop-donation-summary-view

Drop donation summary view

lovelgeorge99 committed:
changed projectTabs adminUserId to true but still not updating in Databse

ae2079 committed:
651 run migration files as js file (#1604)

* Change ormconfig.ts to run different migration files based on process environment name

* Change migration commands of production to use typeorm cli for running migrations js files

* Fix typo in command

lovelgeorge99 committed:
Change for Project Ownship

lovelgeorge99 committed:
Merge branch 'staging' into change-admin#1602

lovelgeorge99 committed:
i have fixed the issue for transfer project ownership in adminBro and fixed for eslint

lovelgeorge99 committed:
Merge branch 'change-admin#1602' of https://github.com/Giveth/impact-graph into change-admin#1602

RamRamez committed:
Revert "651 run migration files as js file (#1604)"

This reverts commit 5140a93e031e1cd5549520c4ac75362c9a011dea.

RamRamez committed:
Merge pull request #1606 from Giveth/revert-#1604

Revert "651 run migration files as js file (#1604)"

mhmdksh committed:
Update staging-pipeline.yml

mhmdksh committed:
Update staging-pipeline.yml

mhmdksh committed:
Update staging-pipeline.yml

divine-comedian committed:
Create funding.json

ae2079 committed:
Fix issue in db migration scripts (#1608)

* Change commands to use typeorm cli

* Select migration files based on node env

* Fix eslint errors on staging branch

RamRamez committed:
add cache to mainCategories query

RamRamez committed:
add cache to findQfRounds query

RamRamez committed:
add cache to findQfRounds query

RamRamez committed:
fix MainCategory test case

lovelgeorge99 committed:
removed unnecessary comments and changed email to admin id

lovelgeorge99 committed:
Merge branch 'staging' of https://github.com/Giveth/impact-graph into change-admin#1602

Meriem-BM committed:
fix: filter unique donors based on passport score and knownAsSybilAddress

RamRamez committed:
set cache duration as env var

RamRamez committed:
Merge pull request #1612 from Giveth/Optimize-qfRounds-mainCategories-query

add caching to QF rounds and mainCategories query

CarlosQ96 committed:
allow refresh concurrently of materialized views (#1611)

Meriem-BM committed:
Merge branch 'staging' of github.com:Giveth/impact-graph into fix/filter_unique_donors

RamRamez committed:
add logger to updateProjectStatistics

Meriem-BM committed:
fix: add new field isDataAnalysisDone to QF Round

Meriem-BM committed:
Merge branch 'staging' into fix/filter_unique_donors

Meriem-BM committed:
fix: refactor condition where data analysis shouldn't be edited

Meriem-BM committed:
Merge branch 'fix/filter_unique_donors' of github.com:Giveth/impact-graph into fix/filter_unique_donors

Meriem-BM committed:
fix: issue while listing recurrring donations

CarlosQ96 committed:
revert concurrent flag for failed views

Meriem-BM committed:
fix: requested change (using subquery to get unique donors count)

Meriem-BM committed:
Merge pull request #1622 from Giveth/fix/recurring_donation_tab

fix: issue while listing recurring donations

Meriem-BM committed:
Merge pull request #1613 from Giveth/fix/filter_unique_donors

fix: filter unique donors based on passport score and knownAsSybilAdd…

CarlosQ96 committed:
add getUserAnalysisScore

CarlosQ96 committed:
revert all materialized views concurrent flag

CarlosQ96 committed:
reactivate concurrent flag for all materialized views

CarlosQ96 committed:
move refresh estimated matching view last

ae2079 committed:
Add some error logs and complete some error messages (#1623)

RamRamez committed:
move refreshProjectEstimatedMatchingView

Meriem-BM committed:
fix: retrieve correct usd value of donations for an archived QF round

RamRamez committed:
skip if there's a wrong tx hash

RamRamez committed:
skip if there's a wrong error in insertDonationsFromQfRoundHistory

mohammadranjbarz committed:
Get giv price from coingecko instead of subgraph (#1621)

* Get giv price from coingecko instead of subgraph

related to Giveth/giveth-dapps-v2#4168

* Fix linter errors

* Remove poignArt stuff

Meriem-BM committed:
Merge pull request #1624 from Giveth/fix/arch_qf_round_usd_val

fix: retrieve correct USD value of donations for an archived QF round

RamRamez committed:
remove unused logs

RamRamez committed:
remove unused logs

RamRamez committed:
force findActiveQfRound to use read replica

RamRamez committed:
rewrite updateProjectStatistics

RamRamez committed:
move refreshProjectEstimatedMatchingView

RamRamez committed:
optimize updateProjectStatistics

RamRamez committed:
Merge pull request #1605 from Giveth/change-admin#1602

changed projectTabs adminUserId to true but still not updating in Database

RamRamez committed:
Merge pull request #1625 from Giveth/optimize-updateProjectStatistics

optimize updateProjectStatistics

RamRamez committed:
ProjectEstimatedMatchingView V3

RamRamez committed:
resolve ProjectEstimatedMatchingView V2 errors

RamRamez committed:
update donationRepository.test.ts

RamRamez committed:
update donationRepository.ts

RamRamez committed:
update project.ts

RamRamez committed:
update ProjectEstimatedMatchingView.ts

RamRamez committed:
update projectRepository.ts

RamRamez committed:
update projectService.ts with new estimated matching structure

RamRamez committed:
update qfRoundHistory.ts with new estimated matching structure

RamRamez committed:
update qfRoundRepository.test.ts with new estimated matching structure

RamRamez committed:
update qfRoundRepository.ts with new estimated matching structure

RamRamez committed:
update qfRoundResolver.ts with new estimated matching structure

RamRamez committed:
update qfUtils.ts with new estimated matching structure

RamRamez committed:
add verified status

RamRamez committed:
add minimumPassportScore to getProjectQfRoundStats

RamRamez committed:
add check for begin and end date to getProjectQfRoundStats

RamRamez committed:
remove refreshProjectEstimatedMatchingView from tests

RamRamez committed:
fix qfRoundRepository.test.ts tests

RamRamez committed:
remove qfRound stats from projects when added or removed from QF round

RamRamez committed:
Merge pull request #1630 from Giveth/fix-estimated-matching-view

Fix Project Estimated Matching View

RamRamez committed:
update added projects only

RamRamez committed:
update added projects only

RamRamez committed:
optimize refreshProjectEstimatedMatchingView

RamRamez committed:
set NODE_ENV for refreshProjectEstimatedMatchingView

RamRamez committed:
merge getQfRoundUniqueDonors and getQfRoundTotalDonations

RamRamez committed:
fix adding valueUsd to query

RamRamez committed:
Merge pull request #1633 from Giveth/merge-getQfRoundUniqueDonors-and-getQfRoundTotalDonations

merge getQfRoundUniqueDonors and getQfRoundTotalDonations

CarlosQ96 committed:
add userQfRoundModelScore

RamRamez committed:
remove minimumPassportScore from getProjectQfRoundStats

RamRamez committed:
remove user join from getProjectQfRoundStats

RamRamez committed:
Merge pull request #1632 from Giveth/optimize-refreshProjectEstimatedMatchingView

optimize refreshProjectEstimatedMatchingView

RamRamez committed:
remove insertDonationsFromQfRoundHistory from syncDonationStatusWithBlockchainNetwork

RamRamez committed:
enable replica

RamRamez committed:
projectVerificationRepository.ts replace .save with update method

RamRamez committed:
projectVerificationRepository.ts replace .save with update method

RamRamez committed:
remove unused fields from logs

RamRamez committed:
remove user from badge warning

RamRamez committed:
optimize projectsWithoutUpdateAfterTimeFrame

CarlosQ96 committed:
fix merge conflicts

mohammadranjbarz committed:
Zkevm integration (#1635)

* Integrate with zkevm chain

related to #1617 #1615 #1618 #1619 #1614 #1620

* Add coingeckoId for wbtc zkevm token

* Fix lint errors

* Remove duplicate keys in provider.ts

* Fix AAve token on zkevm chain

RamRamez committed:
fix merge conflicts

RamRamez committed:
Merge pull request #1636 from Giveth/remove-insertDonationsFromQfRoundHistory-from-syncDonationStatusWithBlockchainNetwork

remove insertDonationsFromQfRoundHistory from syncDonationStatusWitBlockchainNetwork

RamRamez committed:
add projectId to projectRepository.ts

RamRamez committed:
remove unnecessary updatedAt

RamRamez committed:
Merge branch 'staging' into optimize-check-project-verification

Meriem-BM committed:
fix: get unique_donors based on user wallet address not added to sybil table

RamRamez committed:
Merge pull request #1637 from Giveth/optimize-check-project-verification

Optimize check project verification

RamRamez committed:
add cache to findActiveQfRound

mohammadranjbarz committed:
Add on conflicrt do nothing for add zkevm tokens

RamRamez reviewed (approved):
Thanks @mohammadranjbarz

mohammadranjbarz committed:
Fix add polygon zkevm token migration

RamRamez committed:
remove updateTotalDonationsOfProject from migrations

RamRamez committed:
replace updateTotalDonationsOfProject with updateProjectStatistics

RamRamez committed:
replace updateTotalDonationsOfProject with updateProjectStatistics

RamRamez committed:
remove sumDonationValueUsd from project

Meriem-BM committed:
Merge pull request #1643 from Giveth/fix/findArchivedQfRounds

fix: get unique_donors based on user wallet address not added to sybi…

RamRamez committed:
do email verification on project verification form through Ortto

CarlosQ96 committed:
add solana to available networks for qfround

RamRamez committed:
remove knownAsSybilAddress from user

RamRamez committed:
wrong query cache in some functions

RamRamez committed:
Merge pull request #1651 from Giveth/remove-knownAsSybilAddress-from-user

remove knownAsSybilAddress from user

RamRamez committed:
Merge pull request #1652 from Giveth/wrong-query-cache-in-some-functions

wrong query cache in some functions

RamRamez committed:
User Story - Rejected project owner reason

ae2079 committed:
use networkId instead of optimism-only flag (#1653)

* Use networkId instead of optimismOnly flag

* Update graphql queries for tests

ae2079 committed:
fatal errors (#1640)

* change critical error logs to fatal

* add fatal log for donation verification failure with time threshold

kkatusic committed:
added on final step for verification project data

RamRamez committed:
Replace ortto External Embedded Webform for Onboarding guide

Meriem-BM committed:
fix: findArchivedQfRounds query

Meriem-BM committed:
Merge pull request #1660 from Giveth/fix/findArchivedQfRounds

fix: findArchivedQfRounds query

RamRamez committed:
Merge branch 'refs/heads/staging' into add-cache-to-findActiveQfRound

# Conflicts:
#	src/repositories/qfRoundRepository.ts

RamRamez committed:
Merge pull request #1644 from Giveth/add-cache-to-findActiveQfRound

add cache to findActiveQfRound

RamRamez committed:
Merge branch 'staging' into Replace-ortto-External-Embedded-Webform-for-Onboarding-guide

RamRamez committed:
fix subscribeOnboarding args

RamRamez committed:
change output to boolean

RamRamez committed:
Merge branch 'refs/heads/staging' into remove-updateTotalDonationsOfProject

# Conflicts:
#	src/services/projectService.ts

RamRamez committed:
Merge pull request #1648 from Giveth/remove-updateTotalDonationsOfProject

remove updateTotalDonationsOfProject and sumDonationValueUsd

RamRamez committed:
Merge pull request #1649 from Giveth/do-email-verification-on-project-verification-form-through-Ortto

do email verification on project verification form through Ortto

Meriem-BM committed:
WIP: add API key to request & add fetch MBD score record from DB

mohammadranjbarz committed:
Merge branch 'master' into staging

RamRamez committed:
Merge pull request #1654 from Giveth/User-Story---Rejected-project-owner-reason

User Story - Rejected project owner reason

RamRamez committed:
Merge branch 'refs/heads/staging' into Replace-ortto-External-Embedded-Webform-for-Onboarding-guide

# Conflicts:
#	src/adapters/notifications/NotificationCenterAdapter.ts

RamRamez committed:
Merge pull request #1657 from Giveth/Replace-ortto-External-Embedded-Webform-for-Onboarding-guide

Replace ortto External Embedded Webform for Onboarding guide

kkatusic committed:
fixed last step verification

kkatusic committed:
fixed typo

Meriem-BM committed:
WIP: add field to user that holds user MBD score for an active round

RamRamez committed:
disable DB Replica on staging

mohammadranjbarz committed:
Merge branch 'master' into staging

mohammadranjbarz committed:
Merge branch 'staging' of github.com:Giveth/impact-graph into staging

mohammadranjbarz committed:
Merge branch 'master' into staging

mohammadranjbarz committed:
Merge branch 'master' into staging

mohammadranjbarz committed:
Merge branch 'master' into staging

Meriem-BM committed:
fetch data uere with the score from MBD & remove non needed query

mohammadranjbarz committed:
Fix some test cases about stable coin donations

Meriem-BM committed:
fix: issue with adminJs authentication when user has hash as null

Meriem-BM committed:
Merge branch 'staging' into feature_new_qf_scoring_model

Meriem-BM committed:
Merge branch 'staging' of github.com:Giveth/impact-graph into feature_new_qf_scoring_model

Meriem-BM committed:
Merge branch 'feature_new_qf_scoring_model' of github.com:Giveth/impact-graph into feature_new_qf_scoring_model

RamRamez committed:
"Last Update" on project card is not correct

mohammadranjbarz committed:
Fix some test cases about stable coin donations (#1665)

* Fix some test cases about stable coin donations

* Fix few test cases of projectResolver

* Fix few test cases of projectResolver

* Fix test cases of project resolver

* Fix donationResolver test cases

mohammadranjbarz committed:
Merge branch 'staging' of github.com:Giveth/impact-graph into staging

dependabot[bot] committed:
Bump follow-redirects from 1.15.5 to 1.15.6 (#1414)

Bumps [follow-redirects](https://github.com/follow-redirects/follow-redirects) from 1.15.5 to 1.15.6.
- [Release notes](https://github.com/follow-redirects/follow-redirects/releases)
- [Commits](https://github.com/follow-redirects/follow-redirects/compare/v1.15.5...v1.15.6)

---
updated-dependencies:
- dependency-name: follow-redirects
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: mohammadranjbarz <mranjbar.z2993@gmail.com>

dependabot[bot] committed:
Bump @solana/web3.js from 1.87.6 to 1.87.7 (#1492)

Bumps [@solana/web3.js](https://github.com/solana-labs/solana-web3.js) from 1.87.6 to 1.87.7.
- [Release notes](https://github.com/solana-labs/solana-web3.js/releases)
- [Commits](https://github.com/solana-labs/solana-web3.js/commits)

---
updated-dependencies:
- dependency-name: "@solana/web3.js"
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: mohammadranjbarz <mranjbar.z2993@gmail.com>

dependabot[bot] committed:
Bump express from 4.18.2 to 4.19.2 (#1430)

Bumps [express](https://github.com/expressjs/express) from 4.18.2 to 4.19.2.
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/master/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.18.2...4.19.2)

---
updated-dependencies:
- dependency-name: express
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: mohammadranjbarz <mranjbar.z2993@gmail.com>

dependabot[bot] committed:
Bump braces from 3.0.2 to 3.0.3 (#1629)

Bumps [braces](https://github.com/micromatch/braces) from 3.0.2 to 3.0.3.
- [Changelog](https://github.com/micromatch/braces/blob/master/CHANGELOG.md)
- [Commits](https://github.com/micromatch/braces/compare/3.0.2...3.0.3)

---
updated-dependencies:
- dependency-name: braces
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: mohammadranjbarz <mranjbar.z2993@gmail.com>

mohammadranjbarz committed:
Merge branch 'staging' into add_ver_step

mohammadranjbarz committed:
Merge branch 'staging' of github.com:Giveth/impact-graph into staging

mohammadranjbarz committed:
Merge branch 'staging' into feature_new_qf_scoring_model

dependabot[bot] committed:
Bump undici from 5.28.3 to 5.28.4 (#1454)

Bumps [undici](https://github.com/nodejs/undici) from 5.28.3 to 5.28.4.
- [Release notes](https://github.com/nodejs/undici/releases)
- [Commits](https://github.com/nodejs/undici/compare/v5.28.3...v5.28.4)

---
updated-dependencies:
- dependency-name: undici
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: mohammadranjbarz <mranjbar.z2993@gmail.com>

Meriem-BM committed:
Merge pull request #1666 from Giveth/fix/adminJs_auth_issue

fix: issue with adminJs authentication when user has hash as null

RamRamez committed:
Merge branch 'staging' into add_ver_step

RamRamez committed:
Merge branch 'staging' into Last-Update-on-project-card-is-not-correct-#4204-

RamRamez committed:
fix types

kkatusic committed:
Updated project verification form resolver test

fixed previos 3 test with terms and condition update and added one last step test when user complete form and clicked "FINISH" button

RamRamez committed:
fix extracting last comment

ae2079 committed:
get donation to giveth with donation box analytics (#1661)

* Add a method for finding relevant donations to the donationRepository.ts

* Add a method for calculating donations metrics to the donationService.ts

* Add a gql resolver for donations metrics to the donationResolver.ts

* Add unit test for gql query resolver

* Fix division by zero issue

* times 100 average percentage to represent percent

* add useDonationBox field to donation and draftDonation

* fill useDonationBox field with correct data in migrations

* Change donationMetrics endpoint based on change in data schema

* remove unused import

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

* remove unused import

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

* add default value for useDonationBox field

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

* add false removed imports

* add default value for useDonationBox field

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

* fix tests based on changes

* fix tests

* send useDonationBox as last arg and make it optional

* add relevant donation tx hash to donation creation flow

* calculate related donations based on relevant donation tx hash

* add unit tests

* update unit test

* fix bug

* fix bug in donation repository and tests are pass

* Clear All donations after donation metrics test cases

* Add new function for deleting project from db and use it to don't affect other test cases

* comment new test cases to ensure bug is related to that or not

* Clear donations after tests

* Comments new test cases in donationRepository.test.ts

* uncomment tests and Clean up test effects on DB

* Remove project addresses too

* Add a new method to testUtils.ts for removing project from db by id

* change donation dates to now

* Remove clear donations sections

* Change update count to a bigger number to pass test

---------

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

CarlosQ96 committed:
add test for fetching and refreshing qfroundscore for model

kkatusic committed:
Merge pull request #1658 from Giveth/add_ver_step

Added on final step for verification project data and changed terms/condition step

mohammadranjbarz committed:
Merge branch 'staging' of github.com:Giveth/impact-graph into staging

RamRamez committed:
revert changes on package,json

RamRamez committed:
Merge branch 'refs/heads/staging' into Last-Update-on-project-card-is-not-correct-#4204-

RamRamez committed:
revert changes on package,json

RamRamez committed:
fix migration error

RamRamez committed:
fix migration error

RamRamez committed:
Merge pull request #1667 from Giveth/Last-Update-on-project-card-is-not-correct-#4204-

"Last Update" on project card is not correct

Meriem-BM committed:
fix: turn minimumValidUsdValue to nullable when querying

RamRamez committed:
remove updatedAt from projects tab

ae2079 committed:
Remove old donation and add new ones (#1674)

mohammadranjbarz committed:
Merge branch 'staging' of github.com:Giveth/impact-graph into staging

RamRamez committed:
remove verified from projects tab edit menu

RamRamez committed:
remove verified from projectVerificationTab edit menu

CarlosQ96 committed:
fix total donations query

Meriem-BM committed:
Merge branch 'staging' of github.com:Giveth/impact-graph into feature_new_qf_scoring_model

CarlosQ96 committed:
fix archived_round totals

CarlosQ96 committed:
fix latestUpdate column if not exists

CarlosQ96 committed:
fix join for totals

CarlosQ96 committed:
Merge pull request #1675 from Giveth/hotfix_total_donations_query

fix archived_round totals

CarlosQ96 committed:
fix total donations with subquery

CarlosQ96 committed:
Merge branch 'master' into staging

RamRamez committed:
Merge branch 'refs/heads/master' into staging

# Conflicts:
#	src/adapters/superFluid/superFluidAdapter.ts

RamRamez committed:
Merge branch 'refs/heads/master' into staging

# Conflicts:
#	src/server/adminJs/tabs/qfRoundHistoryTab.ts

Meriem-BM committed:
Reslove conflict and merge from `staging` into `feature_new_qf_scoring_model`

Meriem-BM committed:
fix add some error handling and remove commented interface

Meriem-BM committed:
fix: resolve conflict

Meriem-BM committed:
fix: resolve conflict

RamRamez committed:
fix USD value not showing

Meriem-BM committed:
fix: convert timestap retrieved from RPC provider to milisconds

Meriem-BM committed:
fix: donation test cases (milliseconds)

Meriem-BM committed:
fix: donation test cases (milliseconds)

mohammadranjbarz committed:
Endaoment integration (#1663)

* Remove giving block projects that don't have any donation

related to #1599

* Add needed fields to organization entity for endaoment projects

related to Giveth/Roadmap#111

* Remove giving-blocks stuff and endaoment integration thing

* Add categories to endaoment projects

* Fix test cases

* Fix some test cases about stable coin donations

* Add endaoment categories

* Fix remove giving-blocks projects migration

* Add default image for endaoment projects

* Fix linter

* Fix add useDonationBox to draft_donation migration

* Fix migration of adding useDonationBox to donation

* Fix eslint error

* Fix disabling recurring donations for endaoment projects

* Change givingBlock organization to endaoment in pre-test script

* Remove test case of accepted tokens of giving blocks

* Relate all base, optimism and mainnet tokens to endaoment organization

related to #1626

* Fix migration query syntax

Meriem-BM committed:
fix: donation test cases (milliseconds)

mohammadranjbarz committed:
Merge branch 'staging' of github.com:Giveth/impact-graph into staging

Meriem-BM committed:
fix: donation test cases (milliseconds)

Meriem-BM committed:
Merge pull request #1634 from Giveth/feature_new_qf_scoring_model

Feature new qf scoring model

Meriem-BM committed:
Merge branch 'staging' into fix/3456_ethDonation_not_showing

Meriem-BM committed:
Merge pull request #1684 from Giveth/fix/3456_ethDonation_not_showing

Fix: convert timestamp retrieved from RPC provider to milliseconds

mohammadranjbarz committed:
Modify ormconfig for running migrations

mohammadranjbarz committed:
Merge branch 'staging' of github.com:Giveth/impact-graph into staging

mohammadranjbarz committed:
Merge branch 'master' into staging

mohammadranjbarz committed:
Run ts files for migrations

mohammadranjbarz committed:
Make isRecipient of project_addresses of endaoment project true

mohammadranjbarz committed:
Fix type error of one of endaoment projects

RamRamez committed:
add test case for filling prices

RamRamez committed:
Merge pull request #1683 from Giveth/fix-USD-value-not-showing

fix USD value not showing

mohammadranjbarz committed:
Modify add endaoment organization migration to pass the tests

Meriem-BM committed:
fix: convert minimumUserAnalysisScore to Float on qfRound entity

Meriem-BM committed:
fix: revert migration changes

RamRamez committed:
add log for filling prices

Meriem-BM committed:
Merge pull request #1692 from Giveth/fix/minimumUserAnalysisScore

fix: convert minimumUserAnalysisScore to Float on qfRound entity

Reshzera committed:
feat: add github on project socialmedia enum

ae2079 committed:
Merge master to staging (#1696)

* Release relevant donations (#1687)

* get donation to giveth with donation box analytics (#1661)

* Add a method for finding relevant donations to the donationRepository.ts

* Add a method for calculating donations metrics to the donationService.ts

* Add a gql resolver for donations metrics to the donationResolver.ts

* Add unit test for gql query resolver

* Fix division by zero issue

* times 100 average percentage to represent percent

* add useDonationBox field to donation and draftDonation

* fill useDonationBox field with correct data in migrations

* Change donationMetrics endpoint based on change in data schema

* remove unused import

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

* remove unused import

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

* add default value for useDonationBox field

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

* add false removed imports

* add default value for useDonationBox field

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

* fix tests based on changes

* fix tests

* send useDonationBox as last arg and make it optional

* add relevant donation tx hash to donation creation flow

* calculate related donations based on relevant donation tx hash

* add unit tests

* update unit test

* fix bug

* fix bug in donation repository and tests are pass

* Clear All donations after donation metrics test cases

* Add new function for deleting project from db and use it to don't affect other test cases

* comment new test cases to ensure bug is related to that or not

* Clear donations after tests

* Comments new test cases in donationRepository.test.ts

* uncomment tests and Clean up test effects on DB

* Remove project addresses too

* Add a new method to testUtils.ts for removing project from db by id

* change donation dates to now

* Remove clear donations sections

* Change update count to a bigger number to pass test

---------

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

* Remove old donation and add new ones (#1674)

* Remove unused variable

---------

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

* release analytics dashboard dropdown menu related chagnes (#1686)

* use networkId instead of optimism-only flag (#1653)

* Use networkId instead of optimismOnly flag

* Update graphql queries for tests

* Change network id to float in gql schema to fix unit tests

---------

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

mohammadranjbarz committed:
Add organization filter to projectsTab.ts in admin panel

related to https://github.com/Giveth/impact-graph/issues/1599#issuecomment-2233705707

mohammadranjbarz committed:
Merge branch 'staging' of github.com:Giveth/impact-graph into staging

mohammadranjbarz committed:
Fill description summary for endaoment projects

related to https://github.com/Giveth/impact-graph/issues/1599#issuecomment-2233705707

mohammadranjbarz committed:
Fix eslint errors

mohammadranjbarz committed:
Replace special characters of slugs of endaoment projects with -

related to https://github.com/Giveth/giveth-dapps-v2/issues/4280#issuecomment-2233722986

mohammadranjbarz committed:
Fix eslint errors

Meriem-BM committed:
return hardcoded  as user MBD score

Meriem-BM committed:
Merge pull request #1699 from Giveth/fix/minimumUserAnalysisScore

Return hardcoded `70` as user MBD score Temporary

RamRamez committed:
Merge pull request #1694 from Giveth/feat/issue-4216

feat: add github on project socialmedia enum

Meriem-BM committed:
rename qfRound minimumUserAnalysisScore field

Meriem-BM committed:
Merge pull request #1701 from Giveth/fix/minimumUserAnalysisScore

Rename qfRound minimumUserAnalysisScore field

ae2079 committed:
Merge branch 'master' into resolve-conflicts
